fdk/http/event_handler.py

```diff
@@ -0,0 +1,50 @@
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import h11
+
+from fdk import constants
+from fdk import customer_code
+from fdk import log
+from fdk import runner
+
+from fdk.http import routine
+
+
+def event_handle(handle_code: customer_code.Function):
+    """
+    Performs HTTP request-response procedure
+    :param handle_code: customer's code
+    :type handle_code: fdk.customer_code.Function
+    :return: None
+    """
+    async def pure_handler(request_reader, response_writer):
+        log.log("in pure_handler")
+        connection = h11.Connection(h11.SERVER)
+        try:
+            request, body = await routine.read_request(
+                connection, request_reader)
+            headers = dict(request.headers)
+
+            func_response = await runner.handle_request(
+                handle_code, constants.HTTPSTREAM,
+                headers=headers, data=body)
+            log.log("request execution completed")
+
+            await routine.write_response(
+                connection, func_response, response_writer)
+        except Exception as ex:
+            await routine.write_error(ex, connection, response_writer)
```

Comment:
does `write_error` need an additional `try` and `except` in case we fail to write the error out too?